mirdata/mridangam_stroke.py

```diff
@@ -0,0 +1,254 @@
+        sr (float): The sample rate of the audio file
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=44100, mono=True)
+
+
+def download(data_home=None, force_overwrite=False, cleanup=True):
+    """Download the Mridangam Stroke Dataset.
+    Args:
+        data_home (str):
+            Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+        force_overwrite (bool):
+            Whether to overwrite the existing downloaded data
+        cleanup (bool):
+            Whether to delete the zip/tar file after extracting.
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    download_utils.downloader(
+        data_home,
+        remotes=REMOTES,
+        info_message=None,
+        force_overwrite=force_overwrite,
+        cleanup=cleanup,
+    )
+
+
+def validate(data_home=None, silence=False):
+    """Validate if the stored dataset is a valid version
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        missing_files (list): List of file paths that are in the dataset index
+            but missing locally
+        invalid_checksums (list): List of file paths that file exists in the dataset
+            index but has a different checksum compare to the reference checksum
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    missing_files, invalid_checksums = utils.validator(
+        DATA.index, data_home, silence=silence
+    )
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    """Return track ids
+    Returns:
+        (list): A list of track ids
+    """
+    return list(DATA.index.keys())
+
+
+def load(data_home=None):
+    """Load Mridangam Stroke dataset
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        (dict): {`track_id`: track data}
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    data = {}
+    for key in DATA.index.keys():
+        data[key] = Track(key, data_home=data_home)
+    return data
+
+
+def load_stroke_name(audio_path):
+    """Load stroke name of track
+    Args:
+        audio_path (str): Local path where the track is stored.
+    Returns:
+        stroke_name (str): stroke name or type extracted from filename
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+
+    if '.wav' in audio_path:
+        audio_data = audio_path.split('__')[2]
+        stroke_name = audio_data.split('-')[0]
```

Comment:
I have doubts with this
an example of a file name is: 225486__akshaylaya__dheem-c-078.wav 
I think the stroke name here is dheem while the tonic is C. is that correct?
I think it would be better if we validate the stroke names against a pre-defined dictionary of strokes:
`assert stroke_name is in self.stroke_dictionary, "Stroke {} not in stroke dictionary".format(stroke_name)`
where self.stroke_dictionary can be defined in __init__ comprising all the possible strokes